dlib/media/ffmpeg_demuxer.h

```diff
@@ -1034,6 +1034,19 @@ namespace dlib
 
 // ---------------------------------------------------------------------------------------------------
 
+        template <typename image_type>
+        std::enable_if_t<is_image_type<image_type>::value, void>
+        load_frame(image_type& image, const std::string& file_name)
+        {
+            demuxer reader(file_name);
```

Comment:
Here, you do in fact want to set set:
```
decoder_image_args::fmt = pix_traits<pixel_type_t<image_type>>::fmt
```
otherwise `convert()` might fail in:
```
DLIB_ASSERT(f.pixfmt() == pix_traits<pixel>::fmt, "frame doesn't have correct format");
```
By default, `demuxer` converts everything back to RGB. If `image_type` uses BGR, then it will fail. So it's important you tell `demuxer` to convert back to the same pixel format as `image`.

Maybe `demuxer` needs a templated constructor that takes in a pixel type trait for convenience so it can automatically set `decoder_image_args::fmt` to something sensible.

Reply:
Alternatively, we add the function:
```
template<class image_type>
bool demuxer::read(image_type& img);
```
to ```demuxer```. This can automatically convert a `frame` object to a `image_type` and reset the underlying pixel formatter (i.e. the `details::resizer` object)  if required. But then this could potentially override  the ```demuxer::args::args_image``` parameters passed in the constructor.

Reply:
Thank you for the explanation, I think the way you have it now is good already.